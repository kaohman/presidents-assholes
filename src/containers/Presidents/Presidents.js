import React from 'react';
import { connect } from 'react-redux';
import President from '../President/President';

const Presidents = ({ presidents }) => {

  return (
    <div>
      {
        presidents.map(president => {
          return <President president={president} key={president.number}/>
        })
      }
    </div>
  )
}

const mapStateToProps = (state) => ({
  presidents: state.presidents,
});

export default connect(mapStateToProps)(Presidents);