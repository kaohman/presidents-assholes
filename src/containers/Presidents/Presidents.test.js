import { Presidents, mapStateToProps } from './Presidents';

describe('Presidents', () => {

  describe('mapStateToProps', () => {
    it('should return an object with a presidents array', () => {
      const mockState = {
        party: 'Whig',
        presidents: [
          {
            number: 1,
            president: "George Washington",
            birth_year: 1732,
            death_year: 1799,
            took_office: "1789-04-30",
            left_office: "1797-03-04",
            party: "No Party"
          },
          {
            number: 2,
            president: "John Adams",
            birth_year: 1735,
            death_year: 1826,
            took_office: "1797-03-04",
            left_office: "1801-03-04",
            party: "Federalist"
          },
        ],
        otherState: false
      };
      const expected = {
        party: 'Whig',
        presidents: [
          {
            number: 1,
            president: "George Washington",
            birth_year: 1732,
            death_year: 1799,
            took_office: "1789-04-30",
            left_office: "1797-03-04",
            party: "No Party"
          },
          {
            number: 2,
            president: "John Adams",
            birth_year: 1735,
            death_year: 1826,
            took_office: "1797-03-04",
            left_office: "1801-03-04",
            party: "Federalist"
          },
        ]
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});