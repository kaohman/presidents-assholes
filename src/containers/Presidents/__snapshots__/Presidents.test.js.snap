// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presidents should match the snapshot 1`] = `
<div
  className="presidents-container"
/>
`;
